arches/app/search/components/base_search_view.py

```diff
@@ -0,0 +1,163 @@
+#             {
+#                 "componentname": "paging-filter",
+#                 "searchcomponentid": "7aff5819-651c-4390-9b9a-a61221ba52c6",
+#                 "sortorder": 1,
+#             },
+#             {
+#                 "componentname": "provisional-filter",
+#                 "searchcomponentid": "073406ed-93e5-4b5b-9418-b61c26b3640f",
+#                 "sortorder": 2,
+#             },
+#             {
+#                 "componentname": "search-results",
+#                 "searchcomponentid": "00673743-8c1c-4cc0-bd85-c073a52e03ec",
+#                 "sortorder": 3,
+#             }
+#         ]
+#     }
+# }
+
+
+class BaseSearchView(BaseSearchFilter):
+    """
+    Special type of component that specifies which other components to be used,
+    how to execute a search in the search_results method
+    """
+
+    def __init__(self, request=None, user=None, componentname=None):
+        super().__init__(request=request, user=user, componentname=componentname)
+        self.searchview_component = models.SearchComponent.objects.get(
+            componentname=componentname
+        )
+        required_component_sort_order = {
+            item["componentname"]: int(item["sortorder"])
+            for item in self.searchview_component.config["requiredComponents"]
+        }
+        self._required_search_components = list(
+            models.SearchComponent.objects.filter(
+                searchcomponentid__in=[
+                    required_component["searchcomponentid"]
+                    for required_component in self.searchview_component.config[
+                        "requiredComponents"
+                    ]
+                ]
+            )
+        )
+        self._required_search_components = sorted(
+            self._required_search_components,
+            key=lambda item: required_component_sort_order.get(
+                item.componentname, float("inf")
+            ),
+        )
+        available_component_sort_order = {
+            item["componentname"]: int(item["sortorder"])
+            for item in self.searchview_component.config["availableComponents"]
+        }
+        self._available_search_components = list(
+            models.SearchComponent.objects.filter(
+                searchcomponentid__in=[
+                    available_component["searchcomponentid"]
+                    for available_component in self.searchview_component.config[
+                        "availableComponents"
+                    ]
+                ]
+            )
+        )
+        self._available_search_components = sorted(
+            self._available_search_components,
+            key=lambda item: available_component_sort_order.get(
+                item.componentname, float("inf")
+            ),
+        )
+
+    @property
+    def required_search_components(self):
+        return self._required_search_components
+
+    @property
+    def available_search_components(self):
```

Comment:
maybe we should just admit that this is all the filters applied to the search view and call it that
```suggestion
    def search_filters(self):
```

we should also rename the private variables accordingly (eg: _available_search_components --> _search_filters)

Reply:
If we're saying that a search component with a frontend ko component is effectively a filter, what would that make search component without one?

Reply:
I'm really saying that besides the `search-view` their all filters.  Or we should just generally think about them that way.  Strictly speaking most of them are in fact "filters", but other like search export aren't.  I like "filters".  Components is just too generic.